test/test.oauth2.ts

```diff
@@ -1421,6 +1427,88 @@ describe('oauth2', () => {
       assert.strictEqual(params.client_id, 'overridden');
     });
 
+    it('getToken should use basic header auth if provided in options', async () => {
+      const authurl = 'https://sts.googleapis.com/v1/';
+      const basic_auth =
+        'Basic ' +
+        Buffer.from(`${CLIENT_ID}:${CLIENT_SECRET}`).toString('base64');
+      const scope = nock(authurl)
+        .post('/oauthtoken')
+        .matchHeader('authorization', basic_auth)
+        .reply(200, {
+          access_token: 'abc',
+          refresh_token: '123',
+          expires_in: 10,
+        });
+      const opts = {
+        clientId: CLIENT_ID,
+        clientSecret: CLIENT_SECRET,
+        redirectUri: REDIRECT_URI,
+        endpoints: {
+          oauth2AuthBaseUrl: 'https://auth.cloud.google/authorize',
+          oauth2TokenUrl: 'https://sts.googleapis.com/v1/oauthtoken',
+          tokenInfoUrl: 'https://sts.googleapis.com/v1/introspect',
+        },
+        client_authentication: ClientAuthentication.ClientSecretBasic,
+      };
+      const oauth2client = new OAuth2Client(opts);
+      const res = await oauth2client.getToken({
+        code: 'code here',
+        client_id: CLIENT_ID,
+      });
+      scope.done();
+      assert(res.res);
+      assert(res.res.data.access_token === 'abc');
```

Comment:
Improves the error message in the event of a failed assertion (prints expected/actual rather than `false`).

```suggestion
      assert.equal(res.res.data.access_token, 'abc');
```